gluten-ut/spark34/src/test/scala/org/apache/spark/sql/GlutenDynamicPartitionPruningSuite.scala

```diff
@@ -451,17 +483,26 @@ abstract class GlutenDynamicPartitionPruningV1Suite extends GlutenDynamicPartiti
           scanOption.get
         }
 
+        def getDriverMetrics(plan: SparkPlan, key: String): Option[SQLMetric] = {
+          find(plan) { case _: FileSourceScanExec => true }
+            .flatMap(_.asInstanceOf[FileSourceScanExec].driverMetrics.get(key))
+        }
+
         // No dynamic partition pruning, so no static metrics
         // All files in fact table are scanned
         val df1 = sql("SELECT sum(f1) FROM fact")
         df1.collect()
         val scan1 = getFactScan(df1.queryExecution.executedPlan)
         assert(!scan1.metrics.contains("staticFilesNum"))
         assert(!scan1.metrics.contains("staticFilesSize"))
-        val allFilesNum = scan1.metrics("numFiles").value
-        val allFilesSize = scan1.metrics("filesSize").value
-        assert(scan1.metrics("numPartitions").value === numPartitions)
-        assert(scan1.metrics("pruningTime").value === -1)
+        val allFilesNum: Long = getDriverMetrics(scan1, "numFiles").map(_.value).getOrElse(-1)
+        val allFilesSize: Long = getDriverMetrics(scan1, "filesSize").map(_.value).getOrElse(-1)
+        val allPartitions: Long =
+          getDriverMetrics(scan1, "numPartitions").map(_.value).getOrElse(-1)
+        assert(allPartitions === numPartitions)
+        val pruningTimeVal1: Long =
+          getDriverMetrics(scan1, "pruningTime").map(_.value).getOrElse(-1)
+        assert(pruningTimeVal1 === 0)
```

Comment:
why it has changed to 0 instead of -1?

Reply:
Now pruning time comes as default to 0 than -1